libs/event-manager/web-client/data-access/src/lib/event-api.service.ts

```diff
@@ -173,18 +191,46 @@ export class EventApiService {
     });
   }
 
-  findByTicketOwner(owner: PublicKey) {
+  // find published event by firebase id
+  findByFId(eventFId: string): Observable<EventAccount | undefined> {
+    return defer(() => {
+      const reader = this.reader;
+
+      if (reader === null) {
+        return throwError(() => new Error('ProgramReaderMissing'));
+      }
+
+      return from(
+        reader.account.event
+          .all()
+          .then((events) => events as EventAccount[])
+          .then((events: EventAccount[]) =>
+            events.find((event) => event.account.fId === eventFId)
+          )
+      );
+    });
+  }
+
+  // get all the tickets of the connected user
+  findUserTickets() {
     return defer(() => {
       const reader = this.reader;
+      const provider = this.provider;
 
       if (reader === null) {
         return throwError(() => new Error('ProgramReaderMissing'));
       }
 
+      if (provider === null || provider.wallet.publicKey === undefined) {
+        return throwError(
+          () => new Error('Please connect a wallet to see your tickets')
+        );
+      }
+
       return from(
         // get all token accounts from current wallet
         reader.provider.connection
-          .getTokenAccountsByOwner(owner, {
+          .getTokenAccountsByOwner(provider.wallet.publicKey, {
```

Comment:
This query is very expensive. Pretty much everyone uses it, but its still an expensive one. I'd prefer if we could query firebase for this information, but this has some trade-offs that we'll need to discuss.

- If we store the tickets a user owns in our database, what happens if a user transfers a ticket from the outside? 
- How would we deal with "secondary" markets that could re-sell the tickets?

I think we could workaround this if we "freeze" the ticket NFT and have our event as a delegate that can do the transfer on behalf of the user. That way we can ensure all transfers go through our program, hence we can track ticket ownership.